### PR TITLE
Cutting food now moves the sliced pieces!

### PR DESCRIPTION
**Pull request by beck-thompson**
## About the PR

When cutting food into pieces it feels really weird that they are all exactly on the same spot. This just moves them a little bit!

## Media

(Sorry for the bad video it always freeze at the end for no reason...)
https://github.com/space-wizards/space-station-14/assets/107373427/5d2eb1ec-8487-4322-b089-f82efc87680a

## Breaking changes

**Changelog**

:cl: Beck Thompson
- tweak: Cutting food now moves the sliced pieces a small amount!

### CODE REVIEW

File: Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs [hunk 1]
```diff
@@ -10,6 +10,12 @@
 using Robust.Shared.Audio;
 using Robust.Shared.Audio.Systems;
 using Robust.Shared.Containers;
+using Content.Shared.Throwing;
```
Comment (deltanedas) [inline Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:13]: remove

File: Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs [hunk 2]
```diff
@@ -10,6 +10,12 @@
...
+using Robust.Shared.Random;
+using SQLitePCL;
```
Comment (deltanedas) [inline Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:15]: remove

File: Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs [hunk 3]
```diff
@@ -10,6 +10,12 @@
...
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Physics.Systems;
```
Comment (deltanedas) [inline Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:17]: put before random so its alphabetical

File: Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs [hunk 4]
```diff
@@ -10,6 +10,12 @@
...
+using System.Numerics;
```
Comment (deltanedas) [inline Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:18]: remove

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Brandon-Huu** [comment]
The slices sliding around like that is kinda weird

**Comment by beck-thompson** [comment]
I blame VS code for one of those

**Comment by Plykiya** [comment]
does it really need a physics applied to it? just pick a random location within a small space and place it there

**Comment by beck-thompson** [comment]
> does it really need the physics applied to it? just pick a random location within a small space and place it there

I'm not sure how the systems works but I was worried it might spawn inside a wall or something. Maybe it wouldn't be an issue because its such a small distance? That was my reasoning anyway 😆

**Comment by CloneTrooperSimp** [comment]
Personally like it, little things like this always just add more charm to the game to me.

File: Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs [hunk 5]
```diff
@@ -65,6 +70,12 @@ private void OnInteractUsing(Entity<SliceableFoodComponent> entity, ref Interact
             FillSlice(sliceUid, lostSolution);
 
             _audio.PlayPvs(component.Sound, transform.Coordinates, AudioParams.Default.WithVolume(-2));
+
+            // This isn't an inpluse because cuts have very different densities making some get thrown super far..
+            var randVect = _random.NextVector2(2.0f, 2.5f);
+            if (TryComp<PhysicsComponent>(sliceUid, out var physics))
+                _physics.SetLinearVelocity(sliceUid, randVect, body: physics);
```
Comment (EmoGarbage404) [inline Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:77]: this really just ought to be done with the random offset method for the spawn location of the item. applying physics impulses is kinda weird and very small offsets like this aren't going to clip into anything anyways.

Comment (beck-thompson) [reply Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:77]: OK I did it but honestly I don't like it as much! The slices moving around is more fun even if it is cartoonie and less "realistic".

Obviously its up to you and if the PR wont get merged with the velocity I'll change it as its better than before even without velocity.

https://github.com/space-wizards/space-station-14/assets/107373427/ac8fa26c-f6a9-46ef-b427-fe636c70d77b

Comment (beck-thompson) [reply Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:77]: Yeah I asked in the discord and it seems to be split 50/50 on which one is better...

Comment (beck-thompson) [reply Content.Server/Nutrition/EntitySystems/SliceableFoodSystem.cs:77]: I tried asking on discord a while ago but got no response! I'm just going to resolve this for now (I like the physics version more!) but if this wont get merged without doing the random offset let me know and I'll change it 👍

**Comment by beck-thompson** [comment]
Putting this into draft until #30824 is merged!

**Comment by beck-thompson** [comment]
I need to change a lot about the system so I'm going to close and remake the PR so it doesn't get confusing.

**Pull request closed without merging**